### PR TITLE
Bindless Textures

### PR DESCRIPTION
**Pull request by jherico**
This adds support for bindless textures, but does not yet enable them

## Testing

This should have no impact on current rendering, but it does touch the material binding code.  Testing should focus on model objects with textures to verify that they render no differently in this build than in master.  Performance and behavior should be the same.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7742/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7744/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7744/HighFidelity-Beta-PR12566-b7d0260a17654dd79854e5b65058b2cd74c6115f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7751/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7851/app_12566_PR.apk

File: libraries/gpu-gl/src/gpu/gl45/GL45BackendTexture.cpp [hunk 1]
```diff
@@ -20,14 +20,17 @@
...
+#include <gpu/TextureTable.h>
 #include "../gl/GLTexelFormat.h"
 
 using namespace gpu;
 using namespace gpu::gl;
 using namespace gpu::gl45;
 
-#define SPARSE_PAGE_SIZE_OVERHEAD_ESTIMATE 1.3f
 #define MAX_RESOURCE_TEXTURES_PER_FRAME 2
+#define FORCE_STRICT_TEXTURE 0
```
Comment (jherico) [inline libraries/gpu-gl/src/gpu/gl45/GL45BackendTexture.cpp]: A build time option that lets a developer disable variable allocation textures while testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7912/app_12566_PR.apk

**Comment by jherico** [comment]
Requires #12612 to be merged first

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7976/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7976/HighFidelity-Beta-PR12566-3b4e762d4225c78613fd0fe21bc503d12669a114.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7985/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7985/HighFidelity-Beta-PR12566-d2ec60a5b02880b4ea6da5164b69c1452c7ba399.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7991/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7991/HighFidelity-Beta-PR12566-c3fc56bc2c5fcd06e8aa9cd1da671b7407f28cd1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7991/HighFidelity-Beta-PR12566-c3fc56bc2c5fcd06e8aa9cd1da671b7407f28cd1.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7994/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7995/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7995/HighFidelity-Beta-PR12566-c793a8d2c0bf7304a06e4e2b315e7fe03101f3a0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7995/HighFidelity-Beta-PR12566-c793a8d2c0bf7304a06e4e2b315e7fe03101f3a0.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8182/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8182/HighFidelity-Beta-PR12566-ae8df7e24e982735dabd0e50aef3ffabdefe404b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8182/HighFidelity-Beta-PR12566-ae8df7e24e982735dabd0e50aef3ffabdefe404b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8204/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8204/HighFidelity-Beta-PR12566-0e7ee95c954b8f9a0bcfad0234ec846487377537.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8204/HighFidelity-Beta-PR12566-0e7ee95c954b8f9a0bcfad0234ec846487377537.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8386/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8386/HighFidelity-Beta-PR12566-440c01ae05bbe267b6f1b70ce725d3560c8a6fe1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8386/HighFidelity-Beta-PR12566-440c01ae05bbe267b6f1b70ce725d3560c8a6fe1.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Models and objects such as buildings and beach balls rendered and look correct.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8429/app_12566_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8429/HighFidelity-Beta-PR12566-14ea168ab385672448dfa3f028fcbd92c86d05d4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8429/HighFidelity-Beta-PR12566-14ea168ab385672448dfa3f028fcbd92c86d05d4.exe

**Comment by samcake** [comment]
Good to go

**Pull request merged**